src/rust/aggregator/src/data/user_history.rs

```diff
@@ -0,0 +1,411 @@
+use anyhow::anyhow;
+use bigdecimal::{BigDecimal, Zero};
+use chrono::{DateTime, Duration, Utc};
+use sqlx::PgPool;
+
+use super::{Data, DataAggregationError, DataAggregationResult};
+
+#[derive(sqlx::Type, Debug)]
+#[sqlx(type_name = "order_status", rename_all = "lowercase")] // only for PostgreSQL to match a type definition
+pub enum OrderStatus {
+    Open,
+    Closed,
+    Cancelled,
+}
+
+#[derive(sqlx::Type, Debug)]
+#[sqlx(type_name = "order_type", rename_all = "lowercase")] // only for PostgreSQL to match a type definition
+pub enum OrderType {
+    Limit,
+    Market,
+    Swap,
+}
+
+pub struct UserHistory {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl UserHistory {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Data for UserHistory {
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::seconds(5) < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> DataAggregationResult {
+        self.process_and_save_internal().await
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
+        Some(std::time::Duration::from_secs(60 * 60))
```

Comment:
I feel like we're going to want a much lower update interval for this, an hour latency is kind of high. I was thinking 1-5 seconds, it's not like this is a huge amount of work that it's doing.

Reply:
Yes, I intended it to be 5 seconds as that's what I did in the `ready` function but forgot to do it here. Thanks for spotting that